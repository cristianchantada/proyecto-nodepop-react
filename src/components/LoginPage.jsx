import React, { useEffect, useState} from "react";
import { userLogin } from "../api/service";

function LoginPage(){

  const[login, setLogin] = useState(false);
  const[credentials, setCredentials] = useState({
    username: "",
    password: ""
  })

  useEffect(() => {
      console.log(credentials);
      userLogin(credentials).then();
    }
  )

  const handleSubmit = (event) => {
    event.preventDefault();
    console.log(event);
/*     setLogin(true); */
    setCredentials({
        username: event.target.value[0],
        password: event.target.value[1]
      }
    )
  }

  return (
    <div className="container">
      <form onSubmit={handleSubmit}>
        <input name="username" type="text" placeholder="username" />
        <input name="password" type="password" placeholder="password" />
        <button type="submit">Login</button>
      </form>
    </div>
  );
}

export default LoginPage;