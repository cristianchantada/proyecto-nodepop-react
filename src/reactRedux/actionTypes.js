
export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';
export const ADD_TAGS = 'ADDTAGS';
export const ADD_ADVERTS = 'ADD_ADVERTS'