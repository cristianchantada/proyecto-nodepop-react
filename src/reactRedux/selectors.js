
export const getAuth = state => state.auth;

export const getReduxAdverts = state => state.adverts.data;

export const getReduxTags = state => state.tags;

export const getReduxAdvertID = advertId => state => {
    getReduxAdverts(state).find(state => state.id === +advertId);
};

export const getUserInterface = state => state.ui;

export const areAdvertsLoaded = state => state.adverts.areLoaded;