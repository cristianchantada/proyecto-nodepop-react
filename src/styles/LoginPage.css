.container.login {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	height: 65vh;
}

#loginForm {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	background-color: #f2f2f2;
	padding: 30px;
	border-radius: 10px;
	box-shadow: 0 0 10px rgba(0, 0, 0, 0.2);
	max-width: 500px;
	width: 80%;
}

#loginForm input[type="text"],
#loginForm input[type="password"] {
	width: 100%;
	padding: 10px;
	margin-bottom: 15px;
	border: none;
	border-radius: 5px;
	box-shadow: 0 0 5px rgba(0, 0, 0, 0.2);
}

#loginForm label {
	display: flex;
	align-items: center;
	margin-bottom: 15px;
	font-size: 14px;
	color: #666;
}

#loginForm input[type="checkbox"] {
	margin-right: 5px;
}

#loginForm button[type="submit"] {
	background-color: #4caf50;
	color: white;
	padding: 10px 20px;
	border: none;
	border-radius: 5px;
	cursor: pointer;
	transition: all 0.3s ease;
}

#loginForm button[type="submit"]:hover {
	background-color: #3e8e41;
}

.container.login p {
	position: relative;
	color: #ffffff;
	font-size: 20px;
	margin-top: 20px;
	text-align: center;
	width: 100%;
	max-width: 500px;
	background-color: rgba(255, 0, 0, 0.5);
	padding: 10px;
	border-radius: 5px;
	box-shadow: 0 2px 4px rgba(0, 0, 0, 0.4);
	overflow: hidden;
}

.container.login p .delete-icon {
	position: absolute;
	top: 5px;
	right: 5px;
	color: #ffffff;
	font-size: 16px;
	cursor: pointer;
}

@media screen and (max-width: 500px) {
	.container.login p {
		width: calc(100% - 40px);
	}
}
