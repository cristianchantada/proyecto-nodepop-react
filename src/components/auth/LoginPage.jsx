import { useLocation, useNavigate } from "react-router-dom";
import { userLogin } from "../../api/service";
import Layout from "../common/Layout";
import "../../styles/LoginPage.css";
import { useState } from "react";

function LoginPage({ handleLogin }) {
  const navigate = useNavigate();
  const location = useLocation();

  const [error, setError] = useState(null);

  const resetError = () => {
    setError(null);
  };

  const handleSubmit = async (event) => {
    event.preventDefault();

    resetError();

    const credentials = {
      email: event.target.username.value,
      password: event.target.password.value,
    };

    const checked = event.target.checkbox.checked;

    try{
      await userLogin(credentials, checked);
      handleLogin();
      const to = location.state?.from?.pathname || '/';
      navigate(to);
    } catch(error){
      setError(error);
    }
  };

  return (
    <Layout>
      <div className="container login">
        <form id="loginForm" onSubmit={handleSubmit}>
          <input name="username" type="text" placeholder="username" />
          <input name="password" type="password" placeholder="password" />
          <label htmlFor="">
            <input name="checkbox" type="checkbox" />
            ¿Desea guardar la contraseña?
          </label>
          <button type="submit">Login</button>
        </form>
        {error && (
          <p>{error.message}<span onClick={resetError} className="delete-icon">X</span></p>
        )}
      </div>
    </Layout>
  );
}

export default LoginPage;
