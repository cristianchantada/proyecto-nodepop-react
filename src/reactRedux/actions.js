import { LOGIN, LOGOUT, ADD_ADVERTS, ADD_TAGS } from "./actionTypes";

export const login = () => ({
    type: LOGIN
});

export const logout = () => ({
    type: LOGOUT
})

export const tags = () => ({
    type: ADD_TAGS,
    payload: {}
})

export const adverts = () => ({
    type: ADD_ADVERTS,
    payload: {}
})