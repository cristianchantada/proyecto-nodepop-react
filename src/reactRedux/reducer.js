import { LOGIN_REQUEST, LOGIN_FAILURE, LOGIN_SUCCESS, ADD_ADVERTS_SUCCESS, ADD_TAGS_SUCCESS, LOGOUT, USER_INTERFACE_RESET_ERROR } from "./actionTypes";

const defaultState = {
  auth: false,
  adverts: [],
  tags: [],
  userInterface: {
    isLoading: false,
    error: null
  }
}

export function auth(state= defaultState.auth, action) {
  switch(action.type){
    case LOGIN_SUCCESS:
      return true;
    case LOGOUT:
      return false;
    default:
      return state; 
  }
}

export function userInterface(state= defaultState.userInterface, action) {

  if(action.error){
    return {isLoading: false, error: action.payload};
  };

  if(/REQUEST$/.test(action.type)){
    return {isLoading: true, error: null};
  }

  if(/SUCCESS$/.test(action.type)){
    return {isLoading: false, error: null};
  }

  if(action.type === USER_INTERFACE_RESET_ERROR){
    return {...state, error: null}
  }

  return state;
}

export function adverts(state= defaultState.adverts, action){
  if(action.type === ADD_ADVERTS_SUCCESS){
    return action.payload;
  }
    return state;
};


export function tags(state= defaultState.tags, action){
  if(action.type === ADD_TAGS_SUCCESS){
    return action.payload;
  }
    return state;
}

