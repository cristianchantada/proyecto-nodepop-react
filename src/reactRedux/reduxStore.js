import { composeWithDevTools } from '@redux-devtools/extension';
import { createStore } from "redux";
import reducer from "./reducer";
import * as actions from './actions';

const composeEnhancers = composeWithDevTools({
    actions,
});

export default function configureStore (){
    const store = createStore(reducer, composeEnhancers());
    return store;
}
    