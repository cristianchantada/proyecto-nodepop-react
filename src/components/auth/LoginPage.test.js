import { render } from '@testing-library/react';
import LoginPage from './LoginPage';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import {defaultState, userInterface} from '../../reactRedux/reducer';
import { getUserInterface } from '../../reactRedux/selectors';

describe('LoginPage component test', () => {
  
  const store = {
    getState: (error = null) => {
      const state = defaultState;
      state.userInterface = getUserInterface(state);
      state.userInterface.error = error;
      return state;
    },
    subscribe: jest.fn(),
    dispatch: jest.fn()
  }

  const renderComponent = () => render( 
  <Provider store={store}>
    <MemoryRouter>
      <LoginPage />
    </MemoryRouter>
  </Provider>
  );

  test('LoginPage snapshot', () => {
      const {container} = renderComponent();
      expect(container).toMatchSnapshot();
  });   
});