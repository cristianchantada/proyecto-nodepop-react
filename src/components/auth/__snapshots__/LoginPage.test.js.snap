// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component test LoginPage snapshot 1`] = `
<div>
  <div
    class="container layoutContainer"
  >
    <header>
      <a
        class="Link"
        href="/"
      >
        <div
          class="img-title-container"
        >
          <img
            alt="Logo de Node Pop"
            class="logoImg"
            src="node.png"
          />
          <h1>
            NodePop
          </h1>
        </div>
      </a>
      <nav />
    </header>
    <main>
      <div
        class="container login"
      >
        <form
          id="loginForm"
        >
          <input
            name="username"
            placeholder="username"
            type="text"
          />
          <input
            name="password"
            placeholder="password"
            type="password"
          />
          <label
            for=""
          >
            <input
              name="checkbox"
              type="checkbox"
            />
            ¿Desea guardar la contraseña?
          </label>
          <button
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </main>
    <footer>
      <p>
        © Cristian Copyright 2023
      </p>
    </footer>
  </div>
</div>
`;
